### ISSUE TITLE
[Call of Cthulhu: Dark Corners of the Earth] - Windowed Mode brightness issues

### ISSUE THREAD

**Issue by deanff** [elishacloud/dxwrapper#297]
Enabling windowed mode is a god-send as it prevents the game from crashing when alt-tabbing.
Unfortunately, it sets the brightness to 200%, and the in-game brightness menu ceases to function.

FULLSCREEN: https://i.imgur.com/gzowi2P.jpeg
WINDOWED: https://i.imgur.com/qevv4Rh.png

Tested with the latest stable release of dxwrapper on retail edition (same as Steam, I believe) and GOG version.
Let me know if there's anything I can do to help out - and thank you for your efforts on this project.

**Comment by elishacloud** [comment]
That is probably a Gamma issue.  Gamma in Direct3D9 and older does not work in Windowed mode, even fullscreen windowed mode.

**Comment by elishacloud** [comment]
I could use either a shader or Windows gamma.  A shader will take longer.  Using Windows gamma will affect the whole screen not just the game window.

**Comment by deanff** [comment]
It looks like just the game world is affected.
https://i.imgur.com/69LIBPD.png
https://i.imgur.com/D2fLQDl.png

**Comment by elishacloud** [comment]
Ok, try this update.  I recommend using the `FullscreenWindowMode`:  [dxwrapper.zip](https://github.com/user-attachments/files/16984642/dxwrapper.zip)

**Comment by deanff** [comment]
No luck, unfortunately.

Below is what I tested:
FIles copied as is: alt-tab crash
+FullscreenWindowMode: alt-tab crash
+EnableWindowMode: alt-tab crash
-Dd7to9: alt-tab crash
renamed ddraw.dll to d3d9.dll: alt-tab crash
+EnableD3d9Wrapper: same gamma as before. brightness menu has no effect.

Let me know if I've made a mistake somewhere here.

**Comment by elishacloud** [comment]
> Gamma is the same as before, but now the brightness menu works. :)

Which gamma are you referring to that does not work?

> It crashed when alt-tabbing, so I turned on EnableWindowMode.

Is this a bug in the game?  Does it crash without dxwrapper when you do alt+tab?  My Direct3D9 wrapper just returns the calls directly from Direct3D9, so it should not have any issues here.

**Comment by deanff** [comment]
Hi Elisha,
Just tried the latest dxwrapper release and the D3D9 Windowed Mode gamma issue seems to be fixed. Great work!
That said, the in-game brightness menu no longer works when in windowed mode.
Any chance you can have a look at it? (You fixed this pretty quickly last time).

**Comment by elishacloud** [comment]
> That said, the in-game brightness menu no longer works when in windowed mode.

Yes, this is a known issue with Direct3D9.  Supporting Gamma with Direct3D9 in windowed mode [requires either a swapchain](https://learn.microsoft.com/en-us/windows/win32/api/d3d9/nf-d3d9-idirect3ddevice9-setgammaramp) or a pixel shader.  I don't want to use swapchains as this can cause other issues (example with Nvidia's RTX Remix).  At some point I will add a pixel shader for this.  But that won't happen for a while.

**Comment by deanff** [comment]
> At some point I will add a pixel shader for this. But that won't happen for a while.

No worries. Happy to test when it's ready.

**Comment by elishacloud** [comment]
I added a shader for the gamma control.  Here is the new build:  [dxwrapper.zip](https://github.com/user-attachments/files/18405026/dxwrapper.zip)

**Comment by deanff** [comment]
Experienced a lot of rendering issues (walls not showing, blinding lights, etc.)
<img width="960" alt="1" src="https://github.com/user-attachments/assets/f0240e81-f8b3-4920-9e0c-b7692df0024b" />

Previous dxwrapper (Dec 14 Release):
<img width="960" alt="2" src="https://github.com/user-attachments/assets/2b79b1a2-38de-449a-9766-8b080fc41b48" />

From what I could tell, the in-game brightness menu did seem to be working at least.

**Comment by elishacloud** [comment]
Ok, try this one.  Looks like I made a stupid mistake in the last one:  [dxwrapper.zip](https://github.com/user-attachments/files/18406396/dxwrapper.zip)

**Comment by deanff** [comment]
Yep. That did the trick. 
Windowed Mode brightness is as expected, and the in-game brightness menu works.
Thank you for looking into this :)
